reconcile/change_owners/change_log_tracking.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from reconcile.change_owners.bundle import (
+    NoOpFileDiffResolver,
+    QontractServerDiff,
+)
+from reconcile.change_owners.change_owners import fetch_change_type_processors
+from reconcile.change_owners.change_types import ChangeTypeContext
+from reconcile.change_owners.changes import aggregate_file_moves, parse_bundle_changes
+from reconcile.utils import gql
+from reconcile.utils.runtime.integration import NoParams, QontractReconcileIntegration
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.state import init_state
+
+
+class ChangeLogIntegration(QontractReconcileIntegration[NoParams]):
+    def __init__(self) -> None:
+        super().__init__(NoParams())
+        self.qontract_integration = "change-log-tracking"
+        self.qontract_integration_version = make_semver(0, 1, 0)
+
+    @property
+    def name(self) -> str:
+        return self.qontract_integration
+
+    def run(self, dry_run: bool) -> None:
+        change_type_processors = [
+            ctp
+            for ctp in fetch_change_type_processors(
+                gql.get_api(), NoOpFileDiffResolver()
+            )
+            if ctp.labels and "change_log_tracking" in ctp.labels
+        ]
+        state = init_state(
+            integration=self.name,
+        )
+        int_state_path = state.state_path
+        state.state_path = "bundle-archive/diff"
+        change_log: dict[str, list[dict]] = {}
+        for item in state.ls():
+            key = item.lstrip("/")
+            commit = key.rstrip(".json")
+            logging.info(f"Processing commit {commit}")
+            change_log.setdefault(commit, [])
+            obj = state.get(key, None)
+            if not obj:
+                logging.error(f"Error processing commit {commit}")
+                continue
+            diff = QontractServerDiff(**obj)
+            changes = aggregate_file_moves(parse_bundle_changes(diff))
+            for change in changes:
+                logging.debug(f"Processing change {change}")
+                for ctp in change_type_processors:
+                    logging.info(f"Processing change type {ctp.name}")
+                    ctx = ChangeTypeContext(
+                        change_type_processor=ctp,
+                        context="",
+                        origin="",
+                        context_file=change.fileref,
+                        approvers=[],
+                    )
+                    covered_diffs = change.cover_changes(ctx)
+                    if covered_diffs:
+                        change_log[commit].append({
+                            "change_type": ctp.name,
+                        })
```

Comment:
one commit has many changes, one change has many change type processors, can there be duplicate `ctp.name` for the same commit?

Reply:
there can be duplicates indeed. i handled it on the qontract-cli side, but you're right - it's confusing.

in https://github.com/app-sre/qontract-reconcile/pull/4620/commits/54804f652c14645769c3daa6bccbdb5c37d3b738 i've switched to proper classes to make this more readable (and used a set).

Reply:
and some updates in https://github.com/app-sre/qontract-reconcile/pull/4620/commits/9c8bc716ebda6ca25cbe4daebc74afcef3ff85e8

Reply:
Why switch back to `list`? If it's because of json deserialize type issue, can use `BaseModel` instead of `dataclass`, it can handle auto convert list to set.